lib/modules/platform/bitbucket-server/index.ts

```diff
@@ -1154,6 +1156,152 @@ export async function mergePr({
   return true;
 }
 
+export async function expandGroupMembers(
+  reviewers: string[],
+): Promise<string[]> {
+  logger.debug(`expandGroupMembers(${reviewers.join(', ')})`);
+  const expandedUsers: string[] = [];
+
+  for (const reviewer of reviewers) {
+    const [baseEntry, modifier] = reviewer.split(':');
+
+    if (isReviewerGroup(baseEntry)) {
+      const groupName = baseEntry.replace(/^@reviewer-group\//, '');
+      const groupUsers = await getUsersFromReviewerGroup(groupName);
+      if (groupUsers.length === 0) {
+        continue;
+      }
+
+      if (modifier === undefined) {
+        expandedUsers.push(...groupUsers); // Add all users from the group
+        continue;
+      }
+      const randomCount = parseModifier(modifier);
+      if (randomCount === null) {
+        expandedUsers.push(...groupUsers);
+        continue;
+      }
+      expandedUsers.push(...resolveRandom(groupUsers, randomCount));
+    } else {
+      expandedUsers.push(baseEntry); // Add the user entry
+    }
+  }
+
+  return [...new Set(expandedUsers)];
+}
+
+export function extractRulesFromCodeOwnersLines(
+  cleanedLines: string[],
+): FileOwnerRule[] {
+  const results: FileOwnerRule[] = [];
+
+  const reversedLines = cleanedLines
+    .filter((line) => line.trim() !== '' && !line.trim().startsWith('#'))
+    .reverse();
+
+  for (const line of reversedLines) {
+    const safeLine = line.replace(/\\\\ /g, '<<SPACE>>'); // double-backslash before space
```

Comment:
Are you sure that `<<SPACE>>` is never part of the valid input? Would it not be safer to use something like this:
```javascript
function splitEscapedSpaces(str) {
  return str.split(' ').reduce((acc, part) => {
    const prev = acc[acc.length - 1];

    if (prev && prev.endsWith('\\')) {
      acc[acc.length - 1] = prev.slice(0, -1) + ' ' + part;
    } else {
      acc.push(part);
    }

    return acc;
  }, []);
}
```